kernel/kernel-api/src/main/java/io/delta/kernel/internal/data/ScanStateRow.java

```diff
@@ -105,18 +107,20 @@ public static StructType getPhysicalSchema(TableClient tableClient, Row scanStat
      * @return List of partition column names according to the scan state.
      */
     public static List<String> getPartitionColumns(Row scanState) {
```

Comment:
For this and the below, I'm wondering if we can make some of these fields non-nullable? According to the protocol fields `partitionColumns` and `configuration` are always defined in a `Metadata`; otherwise need to accomodate for the null case here

Reply:
>  I'm wondering if we can make some of these fields non-nullable?

How would you do that?

They are defined as `nullable = false` in the schema above already

Reply:
The readers can't (at least at the current moment) enforce this (i.e if you say a field is nullable, they are not going to throw errors if it encounters null values).

I am not sure if this is even enforced on the Delta Spark side.

Reply:
I'm wondering if we can update that schema to be `nullable=false` for those fields for a `ScanStateRow` since according to the protocol they are required in the `Metadata` action --> we shouldn't ever be populating null values when we generate a ScanStateRow.